components/org.wso2.carbon.identity.governance/src/main/java/org/wso2/carbon/identity/governance/store/JDBCIdentityDataStore.java

```diff
@@ -387,6 +387,58 @@ public List<String> listPaginatedUsersNames(List<ExpressionCondition> identityCl
         }
     }
 
+    @Override
+    public List<String> getUserNamesLessThanProvidedClaimValue(String claimURI, String claimValue, int tenantId)
+            throws IdentityException {
+
+        String sqlStmt = SQLQuery.FILTER_USERS_BY_DATA_KEY_LESS_THAN_DATA_VALUE;
+        List<String> userNames = new ArrayList<>();
+        try (Connection connection = IdentityDatabaseUtil.getDBConnection(true)) {
+            try (PreparedStatement prepStmt = connection.prepareStatement(sqlStmt)) {
+                prepStmt.setString(1, claimURI);
+                prepStmt.setInt(2, tenantId);
+                prepStmt.setString(3, claimValue);
+                try (ResultSet resultSet = prepStmt.executeQuery()) {
+                    while (resultSet.next()) {
+                        String username = resultSet.getString(1);
+                        userNames.add(username);
+                    }
+                }
+                IdentityDatabaseUtil.commitTransaction(connection);
+                return userNames;
+            }
+        } catch (SQLException e) {
+            throw new IdentityException("Error occurred while retrieving users from Identity Store.", e);
+        }
+    }
+
+    @Override
+    public  List<String> getUserNamesBetweenProvidedClaimValues(String claimURI, String startValue, String endValue,
+                                                                int tenantId) throws IdentityException {
+
+        String sqlStmt = SQLQuery.FILTER_USERS_BY_DATA_KEY_LESS_THAN_AND_GREATER_THAN_DATA_VALUES;
+        List<String> userNames = new ArrayList<>();
+        try (Connection connection = IdentityDatabaseUtil.getDBConnection(true)) {
+            try (PreparedStatement prepStmt = connection.prepareStatement(sqlStmt)) {
+                prepStmt.setString(1, claimURI);
+                prepStmt.setInt(2, tenantId);
+                prepStmt.setString(3, startValue);
```

Comment:
Please refer my comment on the sql query

Reply:
Fixed with https://github.com/wso2-extensions/identity-governance/pull/726/commits/41e2a3f5a5f65efc1f3d48b4fdece020ae3ece6b